meteo.go

```diff
@@ -0,0 +1,52 @@
+package meteo
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+const (
+	userAgent = "Meteo/0.1 https://github.com/qba73/meteo"
+)
+
+// Weather represents weather conditions
+// in a geographical region.
+type Weather struct {
+	Summary string
+	Temp    float64
+}
+
+// String implements stringer interface.
+func (w Weather) String() string {
+	return fmt.Sprintf("%s %.1f°C", strings.Title(w.Summary), w.Temp)
+}
+
+// NameResolver interface is used by an Meteo Client
+// to obtain geo coordinates for given place located in
+// a country identified by country id.
+type NameResolver interface {
+	// GetCoordinates takes place and country code
+	// and returns geo information like lat and lng.
+	GetCoordinates(placeName, country string) (Place, error)
+}
+
+// RunCLI is a main function that runs the cli machinery.
+func RunCLI() {
+	uname := os.Getenv("GEO_USERNAME")
+	resolver, err := NewWikipediaClient(uname)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
```

Comment:
Should we exit here? Or is it possible to fall back to some other way of resolving place names?

Reply:
This situation needs a bit of thought. If user name is resolved to "", I think we should exit immediately. The reason is that all HTTP calls to the WS provided be ```genomes.org``` should include a valid, registered user name. In our case an empty string would be a special case of not registered user. WS would return status ```401 Unauthorised```.

Reply:
Implemented